### PR TITLE
Only use zend_class_serialize_deny and zend_class_unserialize_deny in PHP versions before 8.1

### PR DESCRIPTION
**Pull request by WyriHaximus**
Refs: https://github.com/amphp/ext-uv/issues/100

### CODE REVIEW

File: php_uv.c [hunk 1]
```diff
@@ -2679,8 +2679,10 @@ static zend_class_entry *php_uv_register_internal_class_ex(const char *name, zen
 	ce.name = zend_new_interned_string(zend_string_init(name, strlen(name), 1));
 	ce.info.internal.builtin_functions = php_uv_empty_methods;
 	new = zend_register_internal_class_ex(&ce, parent);
+#if PHP_VERSION_ID < 80100
 	new->serialize = zend_class_serialize_deny;
 	new->unserialize = zend_class_unserialize_deny;
+#endif
 	new->ce_flags |= ZEND_ACC_FINAL;
```
Comment (trowski) [inline php_uv.c:2686]:
```suggestion
	new->ce_flags |= ZEND_ACC_FINAL;
#if PHP_VERSION_ID < 80100
	new->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE;
#endif
```

File: php_uv.c [hunk 2]
```diff
@@ -2679,9 +2679,13 @@ static zend_class_entry *php_uv_register_internal_class_ex(const char *name, zen
...
 	ce.info.internal.builtin_functions = php_uv_empty_methods;
 	new = zend_register_internal_class_ex(&ce, parent);
+#if PHP_VERSION_ID < 80100
 	new->serialize = zend_class_serialize_deny;
 	new->unserialize = zend_class_unserialize_deny;
-	new->ce_flags |= ZEND_ACC_FINAL;
+#else
+	new->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
+#endif
+    new->ce_flags |= ZEND_ACC_FINAL;
```
Comment (trowski) [inline php_uv.c:2688]: Nit: tab, not spaces.

Comment (WyriHaximus) [reply php_uv.c]: Whoops! Fixed that 👍

File: php_uv.c (hunk 1 above)
Comment (WyriHaximus) [reply php_uv.c:2686]: Done 👍

**Pull request merged**